libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -612,6 +612,20 @@ bool AP_Arming::mission_checks(bool report)
                 return false;
             }
           }
+        AP_Mission::Mission_Command cmd;
+        //get first cmd 
+        if(mission->get_next_nav_cmd(1, cmd) && (_required_mission_items & MIS_ITEM_CHECK_NAV_DIST)) {
```

Comment:
Bad assumption that the second waypoint is a relevant location command.  Check `starts_with_takeoff_command` in `AP_Mission`.